source/common/stats/stat_merger.cc

```diff
@@ -52,6 +47,13 @@ void StatMerger::mergeGauges(const Protobuf::Map<std::string, uint64_t>& gauges)
     }
 
     auto& gauge_ref = temp_scope_->gaugeFromStatName(stat_name, import_mode);
+    if (gauge_ref.importMode() == Gauge::ImportMode::NeverImport) {
```

Comment:
@jmarantz technically, I think this is the race condition. I think it's possible for the child to init as NeverImport after this check,.

Reply:
And actually importMode is actually two flag bits that are set with separate atomic calls. Maybe we can live with this until we refactor, but can you leave a TODO to think about this race and resolve the two-flag-bit setting?

For race-resolution I'm wondering if it's worth fixing; this would only trigger for code-changes across hot-restart. Maybe we can tolerate a possible (but still unlikely) stats glitch for that case?

Reply:
> For race-resolution I'm wondering if it's worth fixing; this would only trigger for code-changes across hot-restart. Maybe we can tolerate a possible (but still unlikely) stats glitch for that case?

It's not just code changes. AFAICT we are always sending NeverImport stats, so it happens even in that case. I will add a comment here and we can decide how/if to fix later IMO.